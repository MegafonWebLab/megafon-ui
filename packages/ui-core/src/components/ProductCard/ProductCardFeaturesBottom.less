@import '~styles/base.less';

@b: e('.mfui-product-card-features-bottom');

@{b} {
    &__list {
        list-style: none;
        margin: 0;
        padding: 0;
    }
    &__title {
        margin: 0 0 8px 0;
    }
    &__item {
        position: relative;
        padding-left: 0;
        & + & {
            margin-top: 9px;
        }
        &_icon {
            padding-left: 41px;
            margin-left: -4px;
        }
    }
    &__item-icon {
        position: absolute;
        top: 0;
        left: 0;
        width: 32px;
        height: 32px;
        fill: @green;
        pointer-events: none;
    }
    &__item-title {
        margin: 0;
    }
}

