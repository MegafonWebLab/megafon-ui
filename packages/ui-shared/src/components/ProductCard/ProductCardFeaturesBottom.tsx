import * as React from 'react';
import * as PropTypes from 'prop-types';
import './ProductCardFeaturesBottom.less';
import { Header, Paragraph, cnCreate } from '@megafon/ui-core';
import { ISecondParamChildren } from './ProductCardFeatures';

interface IProductCardFeaturesBottomProps {
    /** Custom class name */
    className?: string;
    /** Title */
    title?: string;
    /** Showcase params childrens list */
    params: Array<Partial<ISecondParamChildren>>;
}

const cn = cnCreate('mfui-product-card-features-bottom');
class ProductCardFeaturesBottom extends React.Component<IProductCardFeaturesBottomProps, {}> {
    static propTypes = {
        className: PropTypes.string,
        params: PropTypes.array.isRequired,
        title: PropTypes.string,
    };

    static defaultProps: IProductCardFeaturesBottomProps = {
        params: [],
    };

    render() {
        const { params, title, className }: IProductCardFeaturesBottomProps = this.props;

        return (
            <div className={cn('', {}, className)}>
                {title && <Header className={cn('title')} as="h3" margin={false}>{title}</Header>}
                <ul className={cn('list')}>
                    {params.map((param: ISecondParamChildren): React.ReactNode =>
                        <li className={cn('item', { icon: !!param.svgIcon })} key={param.title}>
                            {param.svgIcon && React.cloneElement(param.svgIcon, {
                                className: cn('item-icon'),
                            })}
                            <Header className={cn('item-title')} as="h5" margin={false}>
                                {param.title}
                            </Header>
                            <Paragraph
                                className={cn('item-text')}
                                margin={false}
                                size="small"
                            >
                                {param.caption}
                            </Paragraph>
                        </li>
                    )}
                </ul>
            </div>
        );
    }
}

export default ProductCardFeaturesBottom;
