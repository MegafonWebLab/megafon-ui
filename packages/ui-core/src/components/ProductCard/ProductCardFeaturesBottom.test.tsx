import * as React from 'react';
import { shallow } from 'enzyme';
import ProductCardFeaturesBottom from './ProductCardFeaturesBottom';
import Checked from 'icons/checked_24.svg';

const props = {
    title: 'test',
    params: [
        {
            title: 'test',
            caption: 'test',
            svgIcon: <Checked />,
        },
    ],
};

describe('<ProductCardFeaturesBottom />', () => {
    it('it renders ProductCardFeaturesBottom', () => {
        const wrapper = shallow(<ProductCardFeaturesBottom {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
