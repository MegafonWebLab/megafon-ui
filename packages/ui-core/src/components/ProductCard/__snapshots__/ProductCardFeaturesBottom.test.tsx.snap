// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCardFeaturesBottom /> it renders ProductCardFeaturesBottom 1`] = `
<div
  className="mfui-product-card-features-bottom"
>
  <Header
    as="h3"
    className="mfui-product-card-features-bottom__title"
    color="black"
    margin={false}
  >
    test
  </Header>
  <ul
    className="mfui-product-card-features-bottom__list"
  >
    <li
      className="mfui-product-card-features-bottom__item mfui-product-card-features-bottom__item_icon"
      key="test"
    >
      <test-file-stub
        className="mfui-product-card-features-bottom__item-icon"
      />
      <Header
        as="h5"
        className="mfui-product-card-features-bottom__item-title"
        color="black"
        margin={false}
      >
        test
      </Header>
      <Paragraph
        as="p"
        className="mfui-product-card-features-bottom__item-text"
        color="black"
        marginAll="none"
        sizeAll="small"
        weight="regular"
      >
        test
      </Paragraph>
    </li>
  </ul>
</div>
`;
